src/main/java/org/opensearch/ad/transport/EntityResultRequest.java

```diff
@@ -89,7 +109,26 @@ public long getEnd() {
     public void writeTo(StreamOutput out) throws IOException {
         super.writeTo(out);
         out.writeString(this.detectorId);
-        out.writeMap(entities, (s, e) -> e.writeTo(s), StreamOutput::writeDoubleArray);
+        // guarded with version check. Just in case we send requests to older node where we use String
+        // to represent an entity
+        if (Bwc.supportMultiCategoryFields(out.getVersion())) {
+            out.writeMap(entities, (s, e) -> e.writeTo(s), StreamOutput::writeDoubleArray);
+        } else {
+            Map<String, double[]> oldFormatEntities = new HashMap<>();
+            for (Map.Entry<Entity, double[]> entry : entities.entrySet()) {
+                Map<String, String> attributes = entry.getKey().getAttributes();
+                if (attributes.size() != 1) {
+                    // cannot send a multi-category field entity to old node since it will
+                    // cause EOF exception and stop the detector. Cannot log the entity
+                    // either as there can be a huge number of entities. Since the issue
+                    // is temporary and will be gone after upgrade completes, ignore them.
+                    continue;
+                }
```

Comment:
Can this be logged? Seems this could provide some user confusion if trying to get multi-category results in a mixed cluster.

Reply:
The comment explained why I don't want to log: "Cannot log the entity either as there can be a huge number of entities. Since the issue is temporary and will be gone after upgrade completes, ignore them." Any suggestion on how to log them?

Reply:
I was more suggesting a broader way of logging the problem once per detector - something like "Skipping sending multi-category entities to old node".

Can this be determined outside of this inner `for` loop and logged once? Wouldn't each entity in `entities` have the same number of attributes, so it only has to be determined once?

Reply:
yes, added the log like 

```
Map<String, double[]> oldFormatEntities = new HashMap<>();
            for (Map.Entry<Entity, double[]> entry : entities.entrySet()) {
                Map<String, String> attributes = entry.getKey().getAttributes();
                if (attributes.size() != 1) {
                    // cannot send a multi-category field entity to old node since it will
                    // cause EOF exception and stop the detector. The issue
                    // is temporary and will be gone after upgrade completes.
                    // Since one EntityResultRequest is sent to one node, we can safely
                    // ignore the rest of the requests.
                    LOG.info("Skip sending multi-category entities to an incompatible node. Attributes: ", attributes);
                    break;
                }
                oldFormatEntities.put(entry.getKey().getAttributes().entrySet().iterator().next().getValue(), entry.getValue());
            }
            out.writeMap(oldFormatEntities, StreamOutput::writeString, StreamOutput::writeDoubleArray);
```